phylogeny/prototypeSelection.py

```diff
@@ -511,7 +519,9 @@ def prototype_selection_constructive_maxdist(dm: DistanceMatrix,
     # the element to remove first is the one that has smallest distance to all
     # other. "Removing" works by tagging its distance-sum as infinity. Plus, we
     # decrease the number of available elements by one.
-    minElmIdx = currDists.argmin()
+    minElmIdx = next(x for x in currDists.argsort()
```

Comment:
@josenavas Thank you for reading and thinking of my question! Very valuable comments!

You are right that multiple argsort is what slows down the program. I don't know if argmin can return the index **that is not present in another Python set**. Maybe there is some workaround to achieve this?

@sjanssen2 also thought about the filter > loop idea. However, we discussed and figured that this will generate statistically less robust results. The "include" list of elements serve as a core set, from which the prototypes will grow. Leaving them behind will alter the way prototypes grow.